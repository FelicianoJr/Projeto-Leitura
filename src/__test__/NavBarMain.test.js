import React from "react";
import { shallow } from "enzyme";
import NavbarMain from "../components/NavBarMain";

const setup = (categories = []) => {
  const actions = {
    newPost: jest.fn(),
    sortPost: jest.fn(),
    pushRoute: jest.fn()
  };

  const component = shallow(
    <NavbarMain {...actions} categories={categories} />
  );

  return {
    component: component,
    actions: actions,
    button: component.find("button"),
    select: component.find("select"),
    option: component.find("option"),
    a: component.find("a")
  };
};

describe("<NavBarMain/>", () => {
  it("should render equal text <option>", () => {
    const category = [
      {
        name: "react",
        path: "react"
      },
      {
        name: "redux",
        path: "redux"
      }
    ];
    const { option } = setup(category);
    expect(option.at(0).text()).toEqual("Escolha uma categoria");
    expect(option.at(1).text()).toEqual("react");
    expect(option.at(2).text()).toEqual("redux");
  });

  it("should call sortPost", () => {
    const { actions, select } = setup();
    select.at(1).simulate("change", { target: { value: "100" } });
    expect(actions.sortPost).toBeCalled();
  });

  it("should call PushRoute", () => {
    const { actions, select, component } = setup();
    select.at(0).simulate("change", { target: { value: "100" } });
    expect(actions.pushRoute).toBeCalled();
  });

  it("should button toggle", () => {
    const { button, actions } = setup();
    button.simulate("click");
    expect(actions.newPost).toBeCalled();
  });

  it("shold render one <a>", () => {
    const { a } = setup();
    expect(a.length).toEqual(1);
  });

  it("shold component snapshot", () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
//example change value
// const wrapper = shallow(<Login />);
//    wrapper.find('#password').simulate('change', {target: {name: 'password', value: 'cats'}});

//    expect(wrapper.state('password')).toEqual('cats');
//   })
