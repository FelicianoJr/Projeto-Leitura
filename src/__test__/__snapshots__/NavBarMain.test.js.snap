// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBarMain/> shold component snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavbarMain
    categories={Array []}
    newPost={[Function]}
    pushRoute={[Function]}
    sortPost={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          className="navbar-brand"
          href="/"
>
          Projeto Udacity Leitura
</a>,
        <form
          className="form-inline "
>
          <div
                    className="mr-sm-2"
          >
                    <select
                              className="form-control form-control-sm"
                              defaultValue=""
                              id="category"
                              onChange={[Function]}
                    >
                              <option
                                        disabled={true}
                                        hidden={true}
                                        value=""
                              >
                                        Escolha uma categoria
                              </option>
                    </select>
          </div>
          <div
                    className="mr-sm-2"
          >
                    <select
                              className="form-control form-control-sm"
                              id="sort"
                              onChange={[Function]}
                    >
                              <option
                                        value="SCORE_BIGGER"
                              >
                                        Maior Pontuação
                              </option>
                              <option
                                        value="SCORE_SMALLER"
                              >
                                        Menor Pontuação
                              </option>
                              <option
                                        value="RECENT_POST"
                              >
                                        Postagens Recentes
                              </option>
                              <option
                                        value="OLD_POST"
                              >
                                        Postagens Antigas
                              </option>
                    </select>
          </div>
          <button
                    className="btn btn-outline-info btn-sm"
                    onClick={[Function]}
                    type="button"
          >
                    Nova Postagem
          </button>
</form>,
      ],
      "className": "navbar navbar-expand-lg navbar-dark bg-dark justify-content-between",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Projeto Udacity Leitura",
          "className": "navbar-brand",
          "href": "/",
        },
        "ref": null,
        "rendered": "Projeto Udacity Leitura",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="mr-sm-2"
>
              <select
                            className="form-control form-control-sm"
                            defaultValue=""
                            id="category"
                            onChange={[Function]}
              >
                            <option
                                          disabled={true}
                                          hidden={true}
                                          value=""
                            >
                                          Escolha uma categoria
                            </option>
              </select>
</div>,
            <div
              className="mr-sm-2"
>
              <select
                            className="form-control form-control-sm"
                            id="sort"
                            onChange={[Function]}
              >
                            <option
                                          value="SCORE_BIGGER"
                            >
                                          Maior Pontuação
                            </option>
                            <option
                                          value="SCORE_SMALLER"
                            >
                                          Menor Pontuação
                            </option>
                            <option
                                          value="RECENT_POST"
                            >
                                          Postagens Recentes
                            </option>
                            <option
                                          value="OLD_POST"
                            >
                                          Postagens Antigas
                            </option>
              </select>
</div>,
            <button
              className="btn btn-outline-info btn-sm"
              onClick={[Function]}
              type="button"
>
              Nova Postagem
</button>,
          ],
          "className": "form-inline ",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <select
                className="form-control form-control-sm"
                defaultValue=""
                id="category"
                onChange={[Function]}
>
                <option
                                disabled={true}
                                hidden={true}
                                value=""
                >
                                Escolha uma categoria
                </option>
</select>,
              "className": "mr-sm-2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    disabled={true}
                    hidden={true}
                    value=""
>
                    Escolha uma categoria
</option>,
                  Array [],
                ],
                "className": "form-control form-control-sm",
                "defaultValue": "",
                "id": "category",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Escolha uma categoria",
                    "disabled": true,
                    "hidden": true,
                    "value": "",
                  },
                  "ref": null,
                  "rendered": "Escolha uma categoria",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <select
                className="form-control form-control-sm"
                id="sort"
                onChange={[Function]}
>
                <option
                                value="SCORE_BIGGER"
                >
                                Maior Pontuação
                </option>
                <option
                                value="SCORE_SMALLER"
                >
                                Menor Pontuação
                </option>
                <option
                                value="RECENT_POST"
                >
                                Postagens Recentes
                </option>
                <option
                                value="OLD_POST"
                >
                                Postagens Antigas
                </option>
</select>,
              "className": "mr-sm-2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="SCORE_BIGGER"
>
                    Maior Pontuação
</option>,
                  <option
                    value="SCORE_SMALLER"
>
                    Menor Pontuação
</option>,
                  <option
                    value="RECENT_POST"
>
                    Postagens Recentes
</option>,
                  <option
                    value="OLD_POST"
>
                    Postagens Antigas
</option>,
                ],
                "className": "form-control form-control-sm",
                "id": "sort",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Maior Pontuação",
                    "value": "SCORE_BIGGER",
                  },
                  "ref": null,
                  "rendered": "Maior Pontuação",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Menor Pontuação",
                    "value": "SCORE_SMALLER",
                  },
                  "ref": null,
                  "rendered": "Menor Pontuação",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Postagens Recentes",
                    "value": "RECENT_POST",
                  },
                  "ref": null,
                  "rendered": "Postagens Recentes",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Postagens Antigas",
                    "value": "OLD_POST",
                  },
                  "ref": null,
                  "rendered": "Postagens Antigas",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Nova Postagem",
              "className": "btn btn-outline-info btn-sm",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Nova Postagem",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="navbar-brand"
            href="/"
>
            Projeto Udacity Leitura
</a>,
          <form
            className="form-inline "
>
            <div
                        className="mr-sm-2"
            >
                        <select
                                    className="form-control form-control-sm"
                                    defaultValue=""
                                    id="category"
                                    onChange={[Function]}
                        >
                                    <option
                                                disabled={true}
                                                hidden={true}
                                                value=""
                                    >
                                                Escolha uma categoria
                                    </option>
                        </select>
            </div>
            <div
                        className="mr-sm-2"
            >
                        <select
                                    className="form-control form-control-sm"
                                    id="sort"
                                    onChange={[Function]}
                        >
                                    <option
                                                value="SCORE_BIGGER"
                                    >
                                                Maior Pontuação
                                    </option>
                                    <option
                                                value="SCORE_SMALLER"
                                    >
                                                Menor Pontuação
                                    </option>
                                    <option
                                                value="RECENT_POST"
                                    >
                                                Postagens Recentes
                                    </option>
                                    <option
                                                value="OLD_POST"
                                    >
                                                Postagens Antigas
                                    </option>
                        </select>
            </div>
            <button
                        className="btn btn-outline-info btn-sm"
                        onClick={[Function]}
                        type="button"
            >
                        Nova Postagem
            </button>
</form>,
        ],
        "className": "navbar navbar-expand-lg navbar-dark bg-dark justify-content-between",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Projeto Udacity Leitura",
            "className": "navbar-brand",
            "href": "/",
          },
          "ref": null,
          "rendered": "Projeto Udacity Leitura",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="mr-sm-2"
>
                <select
                                className="form-control form-control-sm"
                                defaultValue=""
                                id="category"
                                onChange={[Function]}
                >
                                <option
                                                disabled={true}
                                                hidden={true}
                                                value=""
                                >
                                                Escolha uma categoria
                                </option>
                </select>
</div>,
              <div
                className="mr-sm-2"
>
                <select
                                className="form-control form-control-sm"
                                id="sort"
                                onChange={[Function]}
                >
                                <option
                                                value="SCORE_BIGGER"
                                >
                                                Maior Pontuação
                                </option>
                                <option
                                                value="SCORE_SMALLER"
                                >
                                                Menor Pontuação
                                </option>
                                <option
                                                value="RECENT_POST"
                                >
                                                Postagens Recentes
                                </option>
                                <option
                                                value="OLD_POST"
                                >
                                                Postagens Antigas
                                </option>
                </select>
</div>,
              <button
                className="btn btn-outline-info btn-sm"
                onClick={[Function]}
                type="button"
>
                Nova Postagem
</button>,
            ],
            "className": "form-inline ",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <select
                  className="form-control form-control-sm"
                  defaultValue=""
                  id="category"
                  onChange={[Function]}
>
                  <option
                                    disabled={true}
                                    hidden={true}
                                    value=""
                  >
                                    Escolha uma categoria
                  </option>
</select>,
                "className": "mr-sm-2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      disabled={true}
                      hidden={true}
                      value=""
>
                      Escolha uma categoria
</option>,
                    Array [],
                  ],
                  "className": "form-control form-control-sm",
                  "defaultValue": "",
                  "id": "category",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Escolha uma categoria",
                      "disabled": true,
                      "hidden": true,
                      "value": "",
                    },
                    "ref": null,
                    "rendered": "Escolha uma categoria",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <select
                  className="form-control form-control-sm"
                  id="sort"
                  onChange={[Function]}
>
                  <option
                                    value="SCORE_BIGGER"
                  >
                                    Maior Pontuação
                  </option>
                  <option
                                    value="SCORE_SMALLER"
                  >
                                    Menor Pontuação
                  </option>
                  <option
                                    value="RECENT_POST"
                  >
                                    Postagens Recentes
                  </option>
                  <option
                                    value="OLD_POST"
                  >
                                    Postagens Antigas
                  </option>
</select>,
                "className": "mr-sm-2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      value="SCORE_BIGGER"
>
                      Maior Pontuação
</option>,
                    <option
                      value="SCORE_SMALLER"
>
                      Menor Pontuação
</option>,
                    <option
                      value="RECENT_POST"
>
                      Postagens Recentes
</option>,
                    <option
                      value="OLD_POST"
>
                      Postagens Antigas
</option>,
                  ],
                  "className": "form-control form-control-sm",
                  "id": "sort",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Maior Pontuação",
                      "value": "SCORE_BIGGER",
                    },
                    "ref": null,
                    "rendered": "Maior Pontuação",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Menor Pontuação",
                      "value": "SCORE_SMALLER",
                    },
                    "ref": null,
                    "rendered": "Menor Pontuação",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Postagens Recentes",
                      "value": "RECENT_POST",
                    },
                    "ref": null,
                    "rendered": "Postagens Recentes",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Postagens Antigas",
                      "value": "OLD_POST",
                    },
                    "ref": null,
                    "rendered": "Postagens Antigas",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Nova Postagem",
                "className": "btn btn-outline-info btn-sm",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Nova Postagem",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
